frontend/lib/resolution/prims.cpp

```diff
@@ -1918,7 +1918,14 @@ CallResolutionResult resolvePrimCall(ResolutionContext* rc,
         auto chplenv = context->getChplEnv();
         auto varName = ci.actual(0).type().param()->toStringParam()->value().str();
         auto it = chplenv->find(varName);
-        auto ret = (it != chplenv->end()) ? it->second : "";
+        std::string ret;
+        if (it != chplenv->end()) {
+          ret = it->second;
+        } else if (varName == "CHPL_DYNO") {
+          ret = "on";
+        } else {
+          context->error(call, "primitive string does not match any environment variable");
+        }
```

Comment:
it's a little scary to me that we're adding this branch for the first time, and that it's specifically for one variable. Have you consider late-populating `chplenv` with this vairable? I think we already have logic for changing the values in the chplenv map as part of `driver.cpp`. It seems reasonable to include this there and then not have to update these two places.

Reply:
As a post-merge cleanup, I'll be looking into a way to implement this more cleanly via driver.cpp and something akin to the ``setChplEnv`` helper.